packages/template-retail-react-app/app/components/_app-config/index.jsx

```diff
@@ -135,6 +135,19 @@ const options = {
                 retry: false
             }
         }
+    },
+    beforeHydrate: (data) => {
+        // Helper for removing the data timestamp.
+        const updateQueryTimeStamp = ({state}) => {
+            delete state.dataUpdatedAt
```

Comment:
🤔 if `dataUpdatedAt` is deleted, how does react-query determine if a query is stale? 

Have you tested it on a query using an arbitrary staleTime value and verify react-query refetch stuff?

I wonder if this could potentially break react-query internals, for example, would the library expects the timestamp to always exist for a completed query? Could it cause exceptions that break the app?

Reply:
Another approach you outlined is to use `Date.now()` and there is a small amount of time between the start of hydration and when PLP starts to fetch data.

How much time would that inaccuracy be? I'd imagine the scale is in tens of milliseconds? Do you want to find out what the exact number is? If the number is super small, maybe it's better to set the time to Date.now()?

Reply:
Both good questions. The date would be created internally in the lib, I can't remember the exact line of file but it will do something like `variable.dataUpdatedAt = data.dataUpdatedAt || anotherValue || Date.now()`. Yes this is private and it could change, which is exactly what I wanted to bring up in this review. The only reason I left this method in the change is because it seems to be more accurate as solving the problem.

"How much time would that inaccuracy be?" I don't think that is something that can be determined, I think it's going to depend on the device of the user. So in that sense it's less accurate leaving us open to having the problem still occur, but at a very rare possibility. 

I can flip it to the Date.now(), which was my first solution, it's probably safer. Do you agree?

Reply:
Yea I agree with fliping to `Date.now()`